engine/src/operators/OrderBy.cpp

```diff
@@ -321,9 +321,187 @@ void process_sort(blazing_frame & input, std::string query_part, Context * query
 	}
 }
 
+
+
+}  // namespace operators
+}  // namespace ral
+
+namespace ral {
+namespace operators {
+namespace experimental {
+
+	using Context = blazingdb::manager::experimental::Context;
+	using Node = blazingdb::transport::experimental::Node;
+	using namespace ral::distribution::experimental;
+
+std::unique_ptr<ral::frame::BlazingTable> process_sort(const ral::frame::BlazingTableView & table, std::string query_part, Context & context) {
+	auto rangeStart = query_part.find("(");
+	auto rangeEnd = query_part.rfind(")") - rangeStart - 1;
+	std::string combined_expression = query_part.substr(rangeStart + 1, rangeEnd);
+
+	std::string limitRowsStr = get_named_expression(combined_expression, "fetch");
+
+	size_t num_sort_columns = count_string_occurrence(combined_expression, "sort");
+
+	std::vector<int8_t> sortOrderTypes(num_sort_columns);
+	std::vector<int> sortColIndices(num_sort_columns);
+	for(int i = 0; i < num_sort_columns; i++) {
+		sortColIndices[i] = get_index(get_named_expression(combined_expression, "sort" + std::to_string(i)));
+		sortOrderTypes[i] =
+			(get_named_expression(combined_expression, "dir" + std::to_string(i)) == DESCENDING_ORDER_SORT_TEXT);		 
+	}
+
+	if(context.getTotalNodes() <= 1) {
+		if(num_sort_columns > 0) {
+			std::unique_ptr<ral::frame::BlazingTable> sorted_table = logicalSort(table, sortColIndices, sortOrderTypes);
+
+			if(!limitRowsStr.empty()) {
+				cudf::size_type limitRows = std::stoi(limitRowsStr);
+				if (limitRows < sorted_table->view().num_rows()){ // lets do the limit only if it will actually do sometihng
+					return logicalLimit(sorted_table->toBlazingTableView(), limitRows);
+				} else {
+					return sorted_table;
+				}
+			} else {
+				return sorted_table;
+			}
+		} else if(!limitRowsStr.empty()) {
+			cudf::size_type limitRows = std::stoi(limitRowsStr);
+			if (limitRows < table.view().num_rows()){ // lets do the limit only if it will actually do sometihng
+				return logicalLimit(table, limitRows);
+			} else {
+				return table.clone();
+			}
+		}
+	} else {
+		if(num_sort_columns > 0) {
```

Comment:
This doesnt seem to make sense, what if num_sort_columns == 0?

Reply:
That is if there was a limit but no sort

Reply:
so what does it do in that case?

Reply:
If there is no sort, it does not call sort. I am not sure what you are asking

Reply:
so you are checking for if(num_sort_columns > 0) {. you still call this function when  num_sort_cols = 0?

Reply:
ok, so that means i cant just use the API we are trying to establish everywhere. Which was BlazingTableView in BlazingTable out.